pandas/tests/test_algos.py

```diff
@@ -1271,19 +1271,38 @@ def test_infinity_sort():
     NegInf = libalgos.NegInfinity()
 
     ref_nums = [NegInf, float("-inf"), -1e100, 0, 1e100, float("inf"), Inf]
+    ref_objects = [NegInf, 'A', 'Ac', 'B', 'C', 'D', 'a', 'z', Inf]
 
     assert all(Inf >= x for x in ref_nums)
     assert all(Inf > x or x is Inf for x in ref_nums)
     assert Inf >= Inf and Inf == Inf
+    assert Inf >= libalgos.Infinity() and Inf == libalgos.Infinity()
     assert not Inf < Inf and not Inf > Inf
+    assert not Inf < libalgos.Infinity() and not Inf > libalgos.Infinity()
+    assert libalgos.Infinity() == libalgos.Infinity()
+    assert not libalgos.Infinity() != libalgos.Infinity()
 
     assert all(NegInf <= x for x in ref_nums)
     assert all(NegInf < x or x is NegInf for x in ref_nums)
     assert NegInf <= NegInf and NegInf == NegInf
+    assert (NegInf <= libalgos.NegInfinity() and
+            NegInf == libalgos.NegInfinity())
     assert not NegInf < NegInf and not NegInf > NegInf
+    assert (not NegInf < libalgos.NegInfinity() and not
+            NegInf > libalgos.NegInfinity())
+    assert libalgos.NegInfinity() == libalgos.NegInfinity()
+    assert not libalgos.NegInfinity() != libalgos.NegInfinity()
+
+    assert Inf > np.nan and Inf >= np.nan and Inf != np.nan
```

Comment:
put the nan comparison in a separate test / section (and comment that you are testing vs nan).

These should be consistent, where *nothing* matches nan.

```
In [1]: np.inf > np.nan
Out[1]: False

In [2]: np.inf < np.nan
Out[2]: False

In [3]: np.inf == np.nan
Out[3]: False

In [4]: np.inf != np.nan
Out[4]: True
```